File: 10_aws-db/1_connection_test.py

import mysql.connector
import configparser

# Read the MySQL configuration from the config file
config = configparser.ConfigParser()
config.read(r'C:\Users\Ivan\Desktop\Python_1\aws-db\config.cfg')

mysql_config = {
    'user': config.get('MYSQL', 'user'),
    'password': config.get('MYSQL', 'password'),
    'host': config.get('MYSQL', 'host'),
}

try:
    # Connect to the MySQL server
    connection = mysql.connector.connect(**mysql_config)
    if connection.is_connected():
        print('Connected to MySQL server')

    # Test the connection
    cursor = connection.cursor()
    cursor.execute('SELECT 1')
    result = cursor.fetchone()
    print('MySQL server version:', result[0])

    # Close the cursor and connection
    cursor.close()
    connection.close()
    print('Disconnected from MySQL server')
except mysql.connector.Error as error:
    print('Error connecting to MySQL server:', error)

File: 10_aws-db/1_connection_test.js
const mysql = require('mysql');
const config = require('./config');

// Read the MySQL configuration from the config file
const mysqlConfig = config.MYSQL;

// Create a MySQL connection pool
const pool = mysql.createPool(mysqlConfig);

// Use the connection pool to query the database
pool.query('SELECT 1', (error, results) => {
  if (error) {
    console.error('Error connecting to MySQL server:', error);
  } else {
    console.log('Connected to MySQL server');
    console.log('MySQL server version:', results[0][0]);
  }
  
  // Close the connection pool
  pool.end();
});